extensions/amp-ad/0.1/amp-ad-api-handler.js

```diff
@@ -122,20 +135,31 @@ export class AmpAdApiHandler {
       this.iframe_.style.visibility = 'hidden';
     }
     listenForOnce(this.iframe_, 'render-start', () => {
+      perf.tick('e_ar');
+      perf.flush();
       if (!this.iframe_) {
         return;
       }
-      if (this.baseInstance_.renderStartResolve_) {
-        this.baseInstance_.renderStartResolve_();
-        this.baseInstance_.renderStartResolve_ = null;
+      if (this.renderStartResolve_) {
+        this.renderStartResolve_();
+        this.renderStartResolve_ = null;
       }
       this.iframe_.style.visibility = '';
     }, this.is3p_);
     this.viewer_.onVisibilityChanged(() => {
       this.sendEmbedInfo_(this.baseInstance_.isInViewport());
     });
     this.element_.appendChild(this.iframe_);
-    return loadPromise(this.iframe_);
+    if (getMode().test) {
+      if (this.renderStartResolve_) {
```

Comment:
This should not exist.
got a bunch of tests failing. i've been working on them for a day, still got one last to fix 😢